### DISCUSSION TITLE
Force space after the function name??

### DISCUSSION THREAD

**Discussion by jvoosten** [PHP-CS-Fixer/PHP-CS-Fixer#7125] [Q&A] [answered]
I've gone over the documentation three times but I can't find any way to force a space after a function name (either declaration or call), e.g.

```
function bla ($v)
{
}


bla ('xyz');
```

There is 'no_spaces_after_function_name' but I want the exact opposite!

**Comment by kubawerlos** [comment]
Hi @jvoosten, is this way a part of some community standards or are you the only person in the universe wanting that space? 😉

**Comment by Wirone** [comment]
@jvoosten there's no such rule, but you're free to create any custom rule you want, and use it in your config 🙂. Fixers that cover edge cases (not commonly agreed by community) have rather small chance to be included in the core.

**Reply by jvoosten** [reply]
"Not commonly agreed"?? Look, without a space code is unreadable. I just want a space *there*, as used in many other projects (not necessarely PHP). 
I will not be the only one using these rules, and if it is not (going to be) part of PHP-CS-Fixer, I don't know how to distribute it to other devs. I simply don't have the time or will to dig into PHP-CS-Fixer to add/program this custom rule that I then need to copy/maintain forever myself. Netbeans can format with spaces, Visual Studio can format with spaces, I just need a command line tool and PHP-CS-Fixer seemed to fit the bill. Too bad, I will search something else then.

**Reply by kubawerlos** [reply]
> without a space code is unreadable

Oh, this argument again... if more spaces means more readable for you, why not this way 👇🏼 ?
```
function    bla    (    $v    )

{


}

```

> I simply don't have the time or will to dig into PHP-CS-Fixer to add/program this custom rule

And you expect someone else to do that for you? When nobody asked for it for over 10 years PHP CS Fixer exists... That's not a good attitude.

**Reply by jvoosten** [reply]
Well, the real downer was "do not expect it to be included into PHP-CS-Fixer". Then I don't want to spend hours digging into PHP-CS-Fixer just for this one thing. I expected it to be there, it's not, it's not going to be there, sorry, goodbye.

**Reply by Wirone** [reply, marked as answer]
Every tool out there (PHPStan, Psalm, PHPCodeSniffer, Rector etc) supports custom rules. That means there's an area for extending built-in behavior for your needs. Maintainers of each of these tools will either accept contribution or tell you that it's outside of their concept and that's OK. As @kubawerlos said - nobody wanted such code style, and there are clear standards in the PHP community:

- [PSR-4](https://www.php-fig.org/psr/psr-2/#43-methods): "Method names MUST NOT be declared with a space after the method name"
- [PSR-12](https://www.php-fig.org/psr/psr-12/#44-methods-and-functions): "Method and function names MUST NOT be declared with space after the method name."
- [PER-CS v1](https://github.com/php-fig/per-coding-style/blob/1.0.0/spec.md#44-methods-and-functions): "Method and function names MUST NOT be declared with space after the method name."
- [PER-CS v2](https://github.com/php-fig/per-coding-style/blob/2.0.0/spec.md#44-methods-and-functions), current one: "Method and function names MUST NOT be declared with space after the method name."

Several people have been working on those documents, thousands of people use it every day. Yes, what you _want_ (which was somehow rude from the beginning) is not something commonly agreed, it's total opposite.

You must understand that every rule comes with maintenance cost. For you, such custom rule wouldn't be much work (you would only need to modify existing `no_spaces_after_function_name` rule and include it in the config). But as a part of the core, with thousands of users, it could lead to issues that would take time from maintainers. We just can't include everything in the core.

> Well, the real downer was "do not expect it to be included into PHP-CS-Fixer".

I did not said that. Please read again.

Last but not least, subjectively speaking, I don't get how lack of that one space makes code unreadable 🤷‍♂️.

**Reply by Wirone** [reply]
@jvoosten please don't unmark my answer, because even though it does not fit your expectations, it covers the topic pretty much and should be easy accessible to other people browsing this discussion.
